sycl/include/sycl/accessor.hpp

```diff
@@ -2972,6 +2974,12 @@ class __SYCL_EBO host_accessor
   }
 
 public:
+  using value_type = typename std::conditional<AccessMode == access_mode::read,
+                                               const DataT, DataT>::type;
+  using reference = value_type &;
+  using const_reference = const DataT &;
+  using size_type = size_t;
```

Comment:
`host_accessor` currently publicly inherit from `accessor<..., access::target::host_buffer>` which should define these. Do we really need these redefined?

Reply:
Thanks @steffenlarsen 
It's added following the 
[Interface for buffer host accessors](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_interface_for_buffer_host_accessors) in specifications.

Reply:
What I mean is, these aliases are defined in the base class (`accessor<..., access::target::host_buffer>`) of `host_accessor` and should already be defined through it. That is, they should be available on `host_accessor` even if they are not added here.

Reply:
@steffenlarsen I understand your point. My impression was that there might have been a reasoning for having them there. I remove it and any other ones declared in base class, from the host_accessor.

Reply:
At some point we'll have to remove the base class, given it is deprecated, but until then it is fine to rely on it. 😄